### DISCUSSION TITLE
Feature request: Make SQL Table structure available to Copilot ! It's extremely empowering to it

### DISCUSSION THREAD

**Discussion by cmp-nct** [community/community#36739] [Copilot Conversations]
Copilot know how to use SQL but he can't do it without knowing the structure.
```
// Insert my employees into bad_employees table
---> copilot has no clue at all what to do, in worst case it will dream up complete garbage tables and columns

However when adding one other comment in the code like this:
// CREATE TABLE `bad_employees` (\n  `id` int NOT NULL,\n  `type` enum('name','location','name_changed') CHARACTER SET latin1 COLLATE latin1_swedish_ci NOT NULL,\n  PRIMARY KEY (`id`,`type`)\n) ENGINE=InnoDB DEFAULT CHARSET=latin1
```

Now suddenly copilot not only knows the syntax, it even chooses the right enum when writing an INSERT statement that actually uses my own implementation routines correctly.

Copilot can really be a very intuitive and powerful tool but it needs to have a relevant scope of data available.
The structure of the database is one of those things it requires to work with database related codes.

**Comment by iamcjwilliams** [comment]
@cmp-nct Thank you for your feedback! 🚀

**Comment by sgbenner** [comment]
I definitely second this! I'm using this tool daily as a data engineer. Its very helpful, but could be 100x more helpful if it knew my data model/schema

**Comment by gregclerk** [comment]
I tried scripting my entire database to a file out of SSMS then I opened that file in a **separate** tab in Azure Data Studio but it didn't seem to help Copilot know the PK-FK joins.  

I even tried Copilot in that **same** tab, but it didn't seem any smarter.  

What am I doing wrong?

**Reply by gregclerk** [reply]
Does it get any smarter by EXECUTING queries or only by having SQL open (even when commented out)?  

I wish I could point it to a folder of .SQL files on disk and have it learn from those.

**Comment by vaibhav3027** [comment]
is there any specific command of comment to point the copilot to my local database schema to use table schemas when suggesting query only from those files.

I mean i have table generation script which has structure of all the tables and want to point copilot to suggest queries according to that schema, any way to do that ?

Also are there any similar commands on which i can train co pilot to do special stuffs on per project basis ?

**Comment by nwheatleKite** [comment]
This would also be enormously helpful to me - to be able to teach Copilot the structure of my database tables and provide it human-descriptive text about the meaning of the fields and their relationships, so it could lean on that information when I - and hundreds of other users - ask it to generate queries to get the exact information they are looking for.
